### PR TITLE
tests: add test to test RDNSS option handling

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This provides a test for the Router Advertisement's RDNSS option to configure an upstream recursive DNS server.

### Testing procedure
Run

    make flash
    sudo make test

for a supported board of your liking.

### Issues/PRs references
See https://github.com/RIOT-OS/RIOT/pull/13732#issuecomment-604739164

### CODE REVIEW

**Comment by benpicco** [comment]
Can you also craft a test with RDNSS and 6ctx - I have a suspicion here… :thinking:

**Comment by miri64** [comment]
> Can you also craft a test with RDNSS and 6ctx - I have a suspicion here… thinking

If you don't tell me your suspicion, I can't really test. Also: Would be tricky, as `scapy`'s support for 6LoWPAN is pretty incomplete and we don't really have a way for host interaction for this (ZEP maybe, but that's only working for `native`). Maybe its best if you dabble in the craft for this yourself ;-).

**Comment by benpicco** [comment]
I see that with `sock_dns` enabled the nodes don't get a global prefix anymore.

**Comment by miri64** [comment]
Added a `Makefile.ci` to the test.

**Comment by benpicco** [comment]
I tried it on `same54-xpro`, this gives

```
sudo /home/benpicco/dev/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200
----> ethos: sending hello.
----> ethos: activating serial pass through.

Traceback (most recent call last):
  File "/home/benpicco/dev/RIOT/tests/gnrc_ipv6_nib_dns/tests/01-run.py", line 82, in <module>
    sys.exit(run(testfunc, timeout=1, echo=True))
  File "/home/benpicco/dev/RIOT/dist/pythonlibs/testrunner/__init__.py", line 30, in run
    testfunc(child)
  File "/home/benpicco/dev/RIOT/tests/gnrc_ipv6_nib_dns/tests/01-run.py", line 66, in testfunc
    lladdr_src = get_host_lladdr(tap)
  File "/home/benpicco/dev/RIOT/tests/gnrc_ipv6_nib_dns/tests/01-run.py", line 44, in get_host_lladdr
    "Can't find host link-local address on interface {}".format(tap)
AssertionError: Can't find host link-local address on interface tap0
make: *** [/home/benpicco/dev/RIOT/tests/gnrc_ipv6_nib_dns/../../Makefile.include:733: test] Error 1
```

I also tried with running `sudo dist/tools/tapsetup/tapsetup` before (If nececary this should be added to the Makefile or Readme), which gives

```
sudo /home/benpicco/dev/RIOT/dist/tools/ethos/ethos tap0 /dev/ttyACM0 115200
----> ethos: sending hello.
----> ethos: activating serial pass through.
ifconfig
Timeout in expect script at "child.expect(r"HWaddr: (?P<hwaddr>[A-Fa-f:0-9]+)\s")" (tests/gnrc_ipv6_nib_dns/tests/01-run.py:68)
```

**Comment by miri64** [comment]
> I also tried with running `sudo dist/tools/tapsetup/tapsetup` before (If nececary this should be added to the Makefile or Readme), which gives

that script at the moment is for `native` only. So no surprise there. I will investigate what's wrong with non-`native` targets.

**Comment by miri64** [comment]
Can reproduce. The other tests using `ethos` don't seem to work either anymore :-/. Will investigate.

**Comment by miri64** [comment]
@benpicco what OS are you using? It seems on Arch Linux, the TAP interface created by `ethos` does not come up on its own.

**Comment by benpicco** [comment]
I'm on Ubuntu 19.10

**Comment by miri64** [comment]
Good thing, that we now can use `start_network.sh` without `uhcpd` ;-).

**Comment by benpicco** [comment]
Now I get

    /home/benpicco/dev/RIOT/dist/tools/ethos/start_network.sh: 68: [: -eq: unexpected operator

**Comment by miri64** [comment]
> Now I get
> 
> ```
> /home/benpicco/dev/RIOT/dist/tools/ethos/start_network.sh: 68: [: -eq: unexpected operator
> ```

Did create_tap fail for you for some reason (this should be better handled, but maybe in a different PR) ;-).

**Comment by benpicco** [comment]
ioctl(TUNSETIFF): Device or resource busy

Oh I already had a tap device set up for a different test - with those deleted I get

```
sudo /home/benpicco/dev/RIOT/dist/tools/ethos/start_network.sh -e /dev/ttyACM1 tap0 "2001:db8::/32" 115200
net.ipv6.conf.tap0.forwarding = 1
net.ipv6.conf.tap0.accept_ra = 0
----> ethos: sending hello.
----> ethos: activating serial pass through.

s
s
```
no response

**Comment by miri64** [comment]
Can you please check if

a) you flashed the right application before hand ;-)
b) you have the same problem on `master`?

With `samr21-xpro` (so UART0_DEV over USB) this works. Maybe your board does not provide UART over USB or something like that.

**Comment by benpicco** [comment]
Yes it's working on `samr21-xpro` and `saml10-xpro` - I must have messed something up there on `same54-xpro` :flushed: 

```
net.ipv6.conf.tap0.forwarding = 1
net.ipv6.conf.tap0.accept_ra = 0
----> ethos: sending hello.
----> ethos: activating serial pass through.
----> ethos: hello reply received
ifconfig

> 
> ifconfig
Iface  5  HWaddr: 00:C5:E5:2B:18:4E 
          L2-PDU:1500 MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::2c5:e5ff:fe2b:184e dns server
 scope: link  VAL
          inet6 group: ff02::1
          inet6 group: ff02::1:ff2b:184e
          
> dns server
DNS server: -.
Sent 1 packets.
dns server

> dns server
DNS server: [2001:db8:affe::dead:beef]:5
Traceback (most recent call last):
  File "/home/benpicco/dev/RIOT/tests/gnrc_ipv6_nib_dns/tests/01-run.py", line 82, in <module>
    sys.exit(run(testfunc, timeout=1, echo=True))
  File "/home/benpicco/dev/RIOT/dist/pythonlibs/testrunner/__init__.py", line 30, in run
    testfunc(child)
  File "/home/benpicco/dev/RIOT/tests/gnrc_ipv6_nib_dns/tests/01-run.py", line 77, in testfunc
    assert dns_server(child) == (DNS_ADDR, DNS_PORT)
AssertionError
```

**Comment by benpicco** [comment]
With only one board connected, the test is working (sometimes).
There seems to be a race condition when the test starts before `tap0` is ready.

`tap0` stays configured after the test, so the second run of the test is usually successful.

**Comment by miri64** [comment]
> `tap0` stays configured after the test, so the second run of the test is usually successful.

Mhhhh it should be removed on `cleanup` in `start_network.sh`...

**Comment by miri64** [comment]
Reverted my last commit as this does not seem to be an `ethos` issue after all. See https://github.com/RIOT-OS/RIOT/pull/13802

**Comment by miri64** [comment]
(confirmed to be working on `iotlab-m3`)

**Comment by benpicco** [comment]
I can confirm the test is working now on `same54-xpro`

<details>
<summary>output</summary>

```
----> ethos: sending hello.
----> ethos: activating serial pass through.
write to fd 3 failed: Invalid argument
----> ethos: hello reply received
ifconfig

> 
> ifconfig
Iface  5  HWaddr: 02:EB:39:55:B4:C3 
          L2-PDU:1500 MTU:1500  HL:64  Source address length: 6
          Link type: wired
          inet6 addr: fe80::eb:39ff:fe55:b4c3  scope: link  VAL
dns server
          inet6 group: ff02::1
          inet6 group: ff02::1:ff55:b4c3
          
> dns server
DNS server: -
.
Sent 1 packets.
dns server
> dns server
DNS server: [2001:db8:affe::dead:beef%5]:53

```
</details>

Please squash!

**Comment by miri64** [comment]
Squashed

File: tests/gnrc_ipv6_nib_dns/tests/01-run.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    lladdr_dst = child.match.group("lladdr").lower()
+
+    assert dns_server(child) is None
+    sendp(Ether(dst=hwaddr_dst) / IPv6(dst=lladdr_dst, src=lladdr_src) /
+          ICMPv6ND_RA() / ICMPv6NDOptRDNSS(dns=[DNS_ADDR]),
+          iface=tap)
+    assert dns_server(child) == (DNS_ADDR, DNS_PORT)
+
+
+
```
Comment (benpicco) [inline tests/gnrc_ipv6_nib_dns/tests/01-run.py:80]: Too many empty lines for `flake8`s liking (just squash directly)

Comment (miri64) [reply tests/gnrc_ipv6_nib_dns/tests/01-run.py:80]: Grmlbrml... vim + middle click + thinkpad scrolling... very bad combination ;-P

**Pull request merged**